### PR TITLE
Regression

### PR DESCRIPTION
**Pull request by SamuelGabriel**
### Change Description

*Try to be precise. You can additionally add comments to your PR, this might help the reviewer a lot.*

I added regression in correspondence to Liam's PR https://github.com/automl/tabpfn-server/pull/49, which should be merged before this PR.

**If you used new dependencies: Did you add them to `requirements.txt`?**
No new dep's.

**Who did you ping on Mattermost to review your PR? Please ping that person again whenever you are ready for another review.**
@liam-sbhoo @davidotte 

## Breaking changes

If you made any breaking changes, please update the version number.
Breaking changes are totally fine, we just need to make sure to keep the users informed and the server in sync.

**Does this PR break the API? If so, what is the corresponding server commit?**
Yes, based on PR https://github.com/automl/tabpfn-server/pull/49

**Does this PR break the user interface? If so, why?**
Yes, I break the user facing API slightly, removing `tabpfn_client.tabpfn_classifier.init/reset` because I think these are more global. They are still available in `tabpfn_client` directly.

---
*Please do not mark comments/conversations as resolved unless you are the assigned reviewer. This helps maintain clarity during the review process.*

### CODE REVIEW

File: tabpfn_client/estimator.py
Comment (SamuelGabriel) [inline tabpfn_client/estimator.py:1]: this is shown as a new file, but the code in TabPFNClassifier (only added doc string), PreprocessorConfig was not touched.

So you would only need to review the TabPFNRegressor bit.

File: tabpfn_client/config.py
Comment (SamuelGabriel) [inline tabpfn_client/config.py:1]: this was simply moved here from tabpfn_classifier, no new logic added

File: tabpfn_client/estimator.py [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+        self.add_fingerprint_features = add_fingerprint_features
+        self.cancel_nan_borders = cancel_nan_borders
+        self.super_bar_dist_averaging = super_bar_dist_averaging
+        self.subsample_samples = subsample_samples
+
+    def fit(self, X, y):
+        # assert init() is called
+        if not config.g_tabpfn_config.is_initialized:
+            raise RuntimeError(
+                "tabpfn_client.init() must be called before using TabPFNClassifier"
```
Comment (liam-sbhoo) [inline tabpfn_client/estimator.py:318]: Should be `TabPFNRegressor` instead.

File: tabpfn_client/estimator.py [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+            )
+
+        if config.g_tabpfn_config.use_server:
+            try:
+                assert (
+                    self.model == "latest_tabpfn_hosted"
+                ), "Only 'latest_tabpfn_hosted' model is supported at the moment for tabpfn_classifier.init(use_server=True)"
```
Comment (liam-sbhoo) [inline tabpfn_client/estimator.py:325]: Should be `TabPFNRegressor` instead.

File: tabpfn_client/estimator.py [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+            except AssertionError as e:
+                print(e)
+            config.g_tabpfn_config.inference_handler.fit(X, y)
+            self.fitted_ = True
+        else:
+            raise NotImplementedError(
+                "Only server mode is supported at the moment for tabpfn_classifier.init(use_server=False)"
```
Comment (liam-sbhoo) [inline tabpfn_client/estimator.py:332]: Should be `TabPFNRegressor` instead.

File: tabpfn_client/client.py [hunk 4]
```diff
@@ -135,7 +142,15 @@ def predict(self, train_set_uid: str, x_test, tabpfn_config: dict | None = None)
 
         self._validate_response(response, "predict")
 
-        return np.array(response.json()["y_pred_proba"])
+        result = response.json()[task]
+
```
Comment (liam-sbhoo) [inline tabpfn_client/client.py:146]: Could you include comment to tell why we handle the result based on the types? i.e. mention we could get classification or regression results. Might be helpful for other developers :)

Comment (liam-sbhoo) [reply tabpfn_client/client.py:146]: OR

Would it be better if we move this logic to `TabPFNClassifier` and `TabPFNRegressor`?

File: tabpfn_client/server_config.yaml
Comment (liam-sbhoo) [inline tabpfn_client/server_config.yaml:1]: I think these changes should not be added.

**Review by liam-sbhoo — COMMENTED** [review]
Thanks for the PR!
Just one thing about the parsing of results (and some minor comments).

p.s. thanks for the comments regarding your changes (e.g. moving code around, renaming and such), it was a great help!

File: tabpfn_client/server_config.yaml
Comment (SamuelGabriel) [reply tabpfn_client/server_config.yaml:1]: sorry! that was a mistake!

File: tabpfn_client/estimator.py (hunk 1 above)
Comment (SamuelGabriel) [reply tabpfn_client/estimator.py]: done

File: tabpfn_client/estimator.py (hunk 3 above)
Comment (SamuelGabriel) [reply tabpfn_client/estimator.py]: done

**Comment by SamuelGabriel** [comment]
I changed all your requested things. Is it ready to be merged (ideally we should merge this and the server's partner PR before the meeting today)

**Comment by liam-sbhoo** [comment]
LTGM, thanks!

**Review by liam-sbhoo — APPROVED** [review]

File: tabpfn_client/client.py
[the remark this answers is not in the document]
Comment (SamuelGabriel) [reply tabpfn_client/client.py:1]: good catch, delted it

**Pull request merged**